lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java

```diff
@@ -892,11 +892,19 @@ public boolean isPageVisited(long pageId, String userId, String owner) {
 		    return true;
 	    }
 
-	    Object [] fields = new Object[2];
-	    fields[0] = Long.toString(pageId);
-	    fields[1] = userId;
```

Comment:
What is the error when you do: `fields[1] = (userId == null)?"":userId;` and don't do anything more.

Reply:
It doesn't throw an exception, just doesn't return any rows.  'ones' is 0 length.

Reply:
But if you use the empty string literal '' then it does return a row.

Reply:
The reason for this is that BasicSqlService in `org.sakaiproject.db.impl.BasicSqlService#prepareStatement` treats empty strings the same as `null`.

Reply:
Yes I saw that - I did 't really want to change that service as its used in lots of places